src/components/editor/Bindings/SchemaInference/Dialog/UpdateSchemaButton.tsx

```diff
@@ -32,10 +30,22 @@ function UpdateSchemaButton({ collectionData, setOpen }: Props) {
     const currentCollection = useResourceConfig_currentCollection();
 
     const originalSchema = useMemo(() => {
+        if (!collectionData || !collectionData.spec) {
+            return {};
+        }
+
         return Object.hasOwn(collectionData.spec, 'readSchema')
             ? collectionData.spec.readSchema
             : collectionData.spec.schema;
-    }, [collectionData.spec]);
+    }, [collectionData]);
+
+    const disableUpdate = useMemo(() => {
+        return (
+            !inferredSpec ||
+            loadingInferredSchema ||
+            isEqual(originalSchema, inferredSpec.readSchema)
+        );
+    }, [inferredSpec, loadingInferredSchema, originalSchema]);
```

Comment:
Moved into `memo` to make code a _bit_ more readable and I feel like the `isEqual` calls might get heavy as specs grow.